From 0ee12981e46d2e6967fd8600e03130b2f43a9604 Mon Sep 17 00:00:00 2001
From: justurn <justurn@live.com>
Date: Sat, 8 Mar 2025 08:57:37 +0800
Subject: [PATCH 1/4] Update mod_accountbound.h

---
 src/mod_accountbound.h | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/mod_accountbound.h b/src/mod_accountbound.h
index 9fcb3f8..664b4b7 100644
--- a/src/mod_accountbound.h
+++ b/src/mod_accountbound.h
@@ -52,11 +52,12 @@ public:
     bool OnGossipHello(Player* /*player*/, Creature* /*creature*/) override;
 
     // PlayerScript
-    void OnAchiComplete(Player* /*player*/, AchievementEntry const* /*achievement*/) override;
-    void OnAfterStoreOrEquipNewItem(Player* /*player*/, uint32 /*vendorslot*/, Item* /*item*/, uint8 /*count*/, uint8 /*bag*/, uint8 /*slot*/, ItemTemplate const* /*pProto*/, Creature* /*pVendor*/, VendorItem const* /*crItem*/, bool /*bStore*/) override;
-    void OnLevelChanged(Player* /*player*/, uint8 /*oldlevel*/) override;
-    void OnLogin(Player* /*player*/) override;
-    void OnSave(Player* /*player*/) override;
+
+    void OnPlayerAchievementComplete(Player* /*player*/, AchievementEntry const* /*achievement*/) override;
+    void OnPlayerBeforeStoreOrEquipNewItem(Player* /*player*/, uint32 /*vendorslot*/, Item* /*item*/, uint8 /*count*/, uint8 /*bag*/, uint8 /*slot*/, ItemTemplate const* /*pProto*/, Creature* /*pVendor*/, VendorItem const* /*crItem*/, bool /*bStore*/) override;
+    void OnPlayerLevelChanged(Player* /*player*/, uint8 /*oldlevel*/) override;
+    void OnPlayerLogin(Player* /*player*/) override;
+    void OnPlayerSave(Player* /*player*/) override;
 
     // WorldScript
     void OnAfterConfigLoad(bool /*reload*/) override;
-- 
2.34.1


From cd792db6112a7644397ee403a1b1ee7a1580390b Mon Sep 17 00:00:00 2001
From: justurn <justurn@live.com>
Date: Sat, 8 Mar 2025 09:04:38 +0800
Subject: [PATCH 2/4] Update mod_accountbound.h

---
 src/mod_accountbound.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mod_accountbound.h b/src/mod_accountbound.h
index 664b4b7..4f82d0a 100644
--- a/src/mod_accountbound.h
+++ b/src/mod_accountbound.h
@@ -54,7 +54,7 @@ public:
     // PlayerScript
 
     void OnPlayerAchievementComplete(Player* /*player*/, AchievementEntry const* /*achievement*/) override;
-    void OnPlayerBeforeStoreOrEquipNewItem(Player* /*player*/, uint32 /*vendorslot*/, Item* /*item*/, uint8 /*count*/, uint8 /*bag*/, uint8 /*slot*/, ItemTemplate const* /*pProto*/, Creature* /*pVendor*/, VendorItem const* /*crItem*/, bool /*bStore*/) override;
+    void OnPlayerBeforeStoreOrEquipNewItem(Player* /*player*/, uint32 /*vendorslot*/, uint32& /*item*/, uint8 /*count*/, uint8 /*bag*/, uint8 /*slot*/, ItemTemplate const* /*pProto*/, Creature* /*pVendor*/, VendorItem const* /*crItem*/, bool /*bStore*/) override;
     void OnPlayerLevelChanged(Player* /*player*/, uint8 /*oldlevel*/) override;
     void OnPlayerLogin(Player* /*player*/) override;
     void OnPlayerSave(Player* /*player*/) override;
-- 
2.34.1


From c74bdb6687f79404b63dea35a536673ae09fbff7 Mon Sep 17 00:00:00 2001
From: justurn <justurn@live.com>
Date: Sat, 8 Mar 2025 13:16:49 +0800
Subject: [PATCH 3/4] Update mod_accountbound_player.cpp

---
 src/mod_accountbound_player.cpp | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/mod_accountbound_player.cpp b/src/mod_accountbound_player.cpp
index 12cfbe3..6839397 100644
--- a/src/mod_accountbound_player.cpp
+++ b/src/mod_accountbound_player.cpp
@@ -1,6 +1,6 @@
 #include "mod_accountbound.h"
 
-void AccountBound::OnAchiComplete(Player* player, AchievementEntry const* achievement)
+void AccountBound::OnPlayerAchievementComplete(Player* player, AchievementEntry const* achievement)
 {
     if (player->IsGameMaster() && !EnableGamemasters)
     {
@@ -16,7 +16,8 @@ void AccountBound::OnAchiComplete(Player* player, AchievementEntry const* achiev
     }
 }
 
-void AccountBound::OnAfterStoreOrEquipNewItem(Player* player, uint32 /*vendorslot*/, Item* item, uint8 /*count*/, uint8 /*bag*/, uint8 /*slot*/, ItemTemplate const* /*pProto*/, Creature* /*pVendor*/, VendorItem const* /*crItem*/, bool /*bStore*/)
+
+void AccountBound::OnPlayerBeforeStoreOrEquipNewItem(Player* player, uint32 /*vendorslot*/, uint32& itemEntry, uint8 /*count*/, uint8 /*bag*/, uint8 /*slot*/, ItemTemplate const* /*pProto*/, Creature* /*pVendor*/, VendorItem const* /*crItem*/, bool /*bStore*/)
 {
     if (player->IsGameMaster() && !EnableGamemasters)
     {
@@ -39,7 +40,7 @@ void AccountBound::OnAfterStoreOrEquipNewItem(Player* player, uint32 /*vendorslo
     }
 }
 
-void AccountBound::OnLevelChanged(Player* player, uint8 /*oldlevel*/)
+void AccountBound::OnPlayerLevelChanged(Player* player, uint8 /*oldlevel*/)
 {
     if (player->IsGameMaster() && !EnableGamemasters)
     {
@@ -52,7 +53,7 @@ void AccountBound::OnLevelChanged(Player* player, uint8 /*oldlevel*/)
     }
 }
 
-void AccountBound::OnLogin(Player* player)
+void AccountBound::OnPlayerLogin(Player* player)
 {
     if (player->IsGameMaster() && !EnableGamemasters)
     {
@@ -70,7 +71,7 @@ void AccountBound::OnLogin(Player* player)
     }
 }
 
-void AccountBound::OnSave(Player* player)
+void AccountBound::OnPlayerSave(Player* player)
 {
     if (player->IsGameMaster() && !EnableGamemasters)
     {
-- 
2.34.1


From f41a90b5b6fad516a9c5c26d27aa84860944fa9f Mon Sep 17 00:00:00 2001
From: justurn <justurn@live.com>
Date: Sat, 8 Mar 2025 13:26:16 +0800
Subject: [PATCH 4/4] Update mod_accountbound_player.cpp

---
 src/mod_accountbound_player.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/mod_accountbound_player.cpp b/src/mod_accountbound_player.cpp
index 6839397..9e14da8 100644
--- a/src/mod_accountbound_player.cpp
+++ b/src/mod_accountbound_player.cpp
@@ -19,6 +19,7 @@ void AccountBound::OnPlayerAchievementComplete(Player* player, AchievementEntry
 
 void AccountBound::OnPlayerBeforeStoreOrEquipNewItem(Player* player, uint32 /*vendorslot*/, uint32& itemEntry, uint8 /*count*/, uint8 /*bag*/, uint8 /*slot*/, ItemTemplate const* /*pProto*/, Creature* /*pVendor*/, VendorItem const* /*crItem*/, bool /*bStore*/)
 {
+    Item * item = player->GetItemByEntry(itemEntry);
     if (player->IsGameMaster() && !EnableGamemasters)
     {
         return;
-- 
2.34.1

