commit 9994610b0b2150d9a34897c22d8e267e846919a3
Author: justurn <justurn@live.com>
Date:   Sun Apr 28 12:44:29 2024 +0800

    Update SpellInfo.cpp

diff --git a/src/server/game/Spells/SpellInfo.cpp b/src/server/game/Spells/SpellInfo.cpp
index c90f3a9c4..c211d96ee 100644
--- a/src/server/game/Spells/SpellInfo.cpp
+++ b/src/server/game/Spells/SpellInfo.cpp
@@ -2621,7 +2621,7 @@ bool SpellInfo::_IsPositiveEffect(uint8 effIndex, bool deep) const
     // Special case: effects which determine positivity of whole spell
     for (uint8 i = 0; i < MAX_SPELL_EFFECTS; ++i)
     {
-        if (Effects[i].IsAura() && Effects[i].ApplyAuraName == SPELL_AURA_MOD_STEALTH)
+        if (Effects[i].IsAura() && (Effects[i].ApplyAuraName == SPELL_AURA_MOD_STEALTH || Effects[i].ApplyAuraName == SPELL_AURA_MANA_SHIELD))
             return true;
     }
 
