commit 72659576674804bc3ddb16515dc5ae29043755fe
Author: justurn <justurn@live.com>
Date:   Fri Apr 26 21:52:28 2024 +0800

    Update Unit.cpp

diff --git a/src/server/game/Entities/Unit/Unit.cpp b/src/server/game/Entities/Unit/Unit.cpp
index 5b1d0d3cc..8766aa63a 100644
--- a/src/server/game/Entities/Unit/Unit.cpp
+++ b/src/server/game/Entities/Unit/Unit.cpp
@@ -7041,6 +7041,35 @@ bool Unit::HandleDummyAuraProc(Unit* victim, uint32 damage, AuraEffect* triggere
                 }
                 switch (dummySpell->Id)
                 {
+                    // Magical Attunement
+                    case 91056:
+                        {
+                            // dont allow non-positive dots to proc
+                            if (!procSpell || !procSpell->IsPositive())
+                                return false;
+
+                            //if healed by another unit (victim)
+                            //if (this == victim)
+                            //    return false;
+
+                            HealInfo const* healInfo = eventInfo.GetHealInfo();
+                            if (!healInfo)
+                            {
+                                return false;
+                            }
+
+                            uint32 effectiveHeal = healInfo->GetEffectiveHeal();
+                            if (effectiveHeal)
+                            {
+                                // heal amount
+                                basepoints0 = int32(CalculatePct(effectiveHeal, triggerAmount));
+                                target = this;
+
+                                if (basepoints0)
+                                    triggered_spell_id = 91055;
+                            }
+                            break;
+                        }
                     // Glyph of Polymorph
                     case 56375:
                         {
@@ -8821,35 +8850,6 @@ bool Unit::HandleAuraProc(Unit* victim, uint32 damage, Aura* triggeredByAura, Sp
                             CastCustomSpell(this, 67545, &bp0, nullptr, nullptr, true, nullptr, triggeredByAura->GetEffect(EFFECT_0), GetGUID());
                             return true;
                         }
-                    // Magical Attunement
-                    case 91056:
-                        {
-                            // dont allow non-positive dots to proc
-                            if (!procSpell || !procSpell->IsPositive())
-                                return false;
-
-                            //if healed by another unit (victim)
-                            //if (this == victim)
-                            //    return false;
-
-                            HealInfo const* healInfo = eventInfo.GetHealInfo();
-                            if (!healInfo)
-                            {
-                                return false;
-                            }
-
-                            uint32 effectiveHeal = healInfo->GetEffectiveHeal();
-                            if (effectiveHeal)
-                            {
-                                // heal amount
-                                basepoints0 = int32(CalculatePct(effectiveHeal, triggerAmount));
-                                target = this;
-
-                                if (basepoints0)
-                                    triggered_spell_id = 91055;
-                            }
-                            break;
-                        }
                 }
                 break;
             }
