commit 79e95c748a4f8019964eab2e86e9e7c77198ff23
Author: justurn <justurn@live.com>
Date:   Sun Oct 8 19:01:54 2023 +0800

    Update boss_twinemperors.cpp

diff --git a/src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_twinemperors.cpp b/src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_twinemperors.cpp
index e527b4076..66666ce18 100644
--- a/src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_twinemperors.cpp
+++ b/src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_twinemperors.cpp
@@ -267,7 +267,7 @@ struct boss_twinemperorsAI : public BossAI
             {
                 if (Creature* twin = GetTwin())
                 {
-                    if (me->IsWithinDist(twin, 60.f))
+                    if (me->IsWithinDist(twin, 10.f))
                         DoCast(twin, SPELL_HEAL_BROTHER, true);
                 }
 
