commit 75c355099c036bbd2d13ffee659778f322f4fcca
Author: StealthyApe <59826804+StealthyApe@users.noreply.github.com>
Date:   Wed Oct 11 19:14:46 2023 +0800

    added cap on mounds

diff --git a/src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_ouro.cpp b/src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_ouro.cpp
index 32d4c51c2..c6b0e8ad7 100644
--- a/src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_ouro.cpp
+++ b/src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_ouro.cpp
@@ -133,7 +133,10 @@ struct boss_ouro : public BossAI
                 })
                 .Schedule(20s, [this](TaskContext context)
                     {
-                        DoCastSelf(SPELL_SUMMON_OURO_MOUNDS, true);
+                        if(_summonedMounds < 10){
+                            DoCastSelf(SPELL_SUMMON_OURO_MOUNDS, true);
+                            _summonedMounds++;
+                        }
                         context.Repeat();
                     });
         }
@@ -269,6 +272,7 @@ struct boss_ouro : public BossAI
         _submergeMelee = 0;
         _submerged = false;
         _enraged = false;
+        _summonedMounds = 0;
     }
 
     void EnterEvadeMode(EvadeReason /*why*/) override
@@ -305,6 +309,7 @@ protected:
     bool _enraged;
     uint8 _submergeMelee;
     bool _submerged;
+    uint8 _summonedMounds;
 
     bool IsPlayerWithinMeleeRange() const
     {
