commit 10fcf70e80a0510257dd8c09e2c2786ceda7ed08
Author: StealthyApe <59826804+StealthyApe@users.noreply.github.com>
Date:   Sat Jun 8 12:13:35 2024 +0800

    Update PlayerUpdates.cpp

diff --git a/src/server/game/Entities/Player/PlayerUpdates.cpp b/src/server/game/Entities/Player/PlayerUpdates.cpp
index 4c1b5bb4a..b9fa31876 100644
--- a/src/server/game/Entities/Player/PlayerUpdates.cpp
+++ b/src/server/game/Entities/Player/PlayerUpdates.cpp
@@ -731,8 +731,14 @@ bool Player::UpdateGatherSkill(uint32 SkillId, uint32 SkillValue,
     // 75-149 - 2 times, 225-299 - 8 times
     switch (SkillId)
     {
-    case SKILL_HERBALISM:
     case SKILL_LOCKPICKING:
+    return UpdateSkillPro(SkillId,
+                    SkillGainChance(SkillValue, RedLevel + 100,
+                    RedLevel + 50, RedLevel + 25) *
+                    Multiplicator,
+                    specialLockingBusiness);
+    break;
+    case SKILL_HERBALISM:
     case SKILL_JEWELCRAFTING:
     case SKILL_INSCRIPTION:
         return UpdateSkillPro(SkillId,
