commit ad144e69c492a1cdbbccb5ceba0161a3841b3896
Author: justurn <justurn@live.com>
Date:   Sat Mar 8 08:05:31 2025 +0800

    Update PlayerUpdates.cpp

diff --git a/src/server/game/Entities/Player/PlayerUpdates.cpp b/src/server/game/Entities/Player/PlayerUpdates.cpp
index 1e6a9c582..b1af327f8 100644
--- a/src/server/game/Entities/Player/PlayerUpdates.cpp
+++ b/src/server/game/Entities/Player/PlayerUpdates.cpp
@@ -770,10 +770,10 @@ bool Player::UpdateGatherSkill(uint32 SkillId, uint32 SkillValue,
     uint32 specialLockingBusiness =sWorld->getIntConfig(CONFIG_SKILL_GAIN_CRAFTING);
 
     if(SkillId == SKILL_LOCKPICKING){
-        sScriptMgr->OnUpdateGatheringSkill(this, SkillId, SkillValue, RedLevel + 100, RedLevel + 50, RedLevel + 25, specialLockingBusiness);
+        sScriptMgr->OnPlayerUpdateGatheringSkill(this, SkillId, SkillValue, RedLevel + 100, RedLevel + 50, RedLevel + 25, specialLockingBusiness);
     }
     else {
-        sScriptMgr->OnUpdateGatheringSkill(this, SkillId, SkillValue, RedLevel + 100, RedLevel + 50, RedLevel + 25, gathering_skill_gain);
+        sScriptMgr->OnPlayerUpdateGatheringSkill(this, SkillId, SkillValue, RedLevel + 100, RedLevel + 50, RedLevel + 25, gathering_skill_gain);
     }
 
     // For skinning and Mining chance decrease with level. 1-74 - no decrease,
