commit 575cac1e4b6071e74775f3c462632b120b3f9af8
Author: justurn <justurn@live.com>
Date:   Sun May 21 12:41:58 2023 +0800

    Update boss_viscidus.cpp
    
    nerf counters on viscidus

diff --git a/src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_viscidus.cpp b/src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_viscidus.cpp
index b39a94a8d..f44aba668 100644
--- a/src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_viscidus.cpp
+++ b/src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_viscidus.cpp
@@ -71,13 +71,13 @@ enum Emotes
 
 enum HitCounter
 {
-    HITCOUNTER_SLOW             = 100,
-    HITCOUNTER_SLOW_MORE        = 150,
-    HITCOUNTER_FREEZE           = 200,
+    HITCOUNTER_SLOW             = 10,
+    HITCOUNTER_SLOW_MORE        = 15,
+    HITCOUNTER_FREEZE           = 20,
 
-    HITCOUNTER_CRACK            = 50,
-    HITCOUNTER_SHATTER          = 100,
-    HITCOUNTER_EXPLODE          = 150,
+    HITCOUNTER_CRACK            = 10,
+    HITCOUNTER_SHATTER          = 20,
+    HITCOUNTER_EXPLODE          = 30,
 };
 
 enum MovePoints
