commit 0a0bdb23dcf1cdc90a9aec819f2cfa781775d674
Author: justurn <justurn@live.com>
Date:   Fri Jun 14 22:07:31 2024 +0800

    Update boss_nexusprince_shaffar.cpp

diff --git a/src/server/scripts/Outland/Auchindoun/ManaTombs/boss_nexusprince_shaffar.cpp b/src/server/scripts/Outland/Auchindoun/ManaTombs/boss_nexusprince_shaffar.cpp
index 07f5762e5..6ac9d99a9 100644
--- a/src/server/scripts/Outland/Auchindoun/ManaTombs/boss_nexusprince_shaffar.cpp
+++ b/src/server/scripts/Outland/Auchindoun/ManaTombs/boss_nexusprince_shaffar.cpp
@@ -89,14 +89,16 @@ struct boss_nexusprince_shaffar : public BossAI
         _JustEngagedWith();
         Talk(SAY_AGGRO);
         summons.DoZoneInCombat();
-        scheduler.Schedule(10s, [this](TaskContext context)
+        // reduce frequency to from 10s to 20s
+        scheduler.Schedule(20s, [this](TaskContext context) 
         {
             if (!urand(0, 3))
             {
                 Talk(SAY_SUMMON);
             }
             DoCastSelf(SPELL_ETHEREAL_BEACON, true);
-            context.Repeat(10s);
+            context.Repeat(20s);
+            // reduce frequency to from 10s to 20s
         }).Schedule(4s, [this](TaskContext context)
         {
             DoCastVictim(RAND(SPELL_FROSTBOLT, SPELL_FIREBALL));
