commit ab7c8603404d3933a64b6d73027c547c7054da8b
Author: justurn <justurn@live.com>
Date:   Tue Jun 17 08:03:49 2025 +0800

    update timezone

diff --git a/apps/docker/Dockerfile b/apps/docker/Dockerfile
index e003ae875..d6011395e 100644
--- a/apps/docker/Dockerfile
+++ b/apps/docker/Dockerfile
@@ -1,5 +1,5 @@
 ARG UBUNTU_VERSION=22.04 # lts
-ARG TZ=Etc/UTC
+ARG TZ=Australia/Perth
 
 # This target lays out the general directory skeleton for AzerothCore,
 # This target isn't intended to be directly used
